File: exercise/python/day08/domain/toy.py
from enum import Enum


class State(Enum):
    UNASSIGNED = "UNASSIGNED"
    IN_PRODUCTION = "IN_PRODUCTION"
    COMPLETED = "COMPLETED"


class Toy:
    def __init__(self, name: str, state: State):
        self.name = name
        self.state = state


class ToyRepository:
    def find_by_name(self, name: str) -> Toy:
        raise NotImplementedError

    def save(self, toy: Toy):
        raise NotImplementedError

File: exercise/ts/day08/src/domain/toy.ts
export class Toy {
    static State = {
        UNASSIGNED: 'UNASSIGNED',
        IN_PRODUCTION: 'IN_PRODUCTION',
        COMPLETED: 'COMPLETED'
    };

    private readonly name: string;
    private state: string;

    constructor(name: string, state: string) {
        this.name = name;
        this.state = state;
    }

    getName(): string {
        return this.name;
    }

    getState(): string {
        return this.state;
    }

    setState(state: string): void {
        this.state = state;
    }
}